task-tracker-webapp/src/main/java/com/isa/tasktrackerwebapp/model/JsonData.java

```diff
@@ -0,0 +1,53 @@
+package com.isa.tasktrackerwebapp.model;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JsonData {
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+    private static final File userFile = new File("src/main/resources/database/users.json");
+    private static List<User> users = new ArrayList<>();
+
+    private JsonData() {
+    }
+
+    public static void updateUserData(User user) {
+        users.addAll(getUsers());
+        saveUserData(users.stream()
+                .map(u -> u.equals(user) ? user : u)
+                .toList());
+    }
+
+    public static void saveNewUser(User user) {
+        users.addAll(getUsers());
+        users.add(user);
+        saveUserData(users);
+    }
+
+    private static List<User> readUsers() {
+        try {
+            users = objectMapper.readValue(userFile, new TypeReference<>() {
+            });
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return users;
+    }
+
+    public static List<User> getUsers() {
+        users.addAll(readUsers());
+        return users;
+    }
```

Comment:
tego trochę nie kumam i podobnie macie w taskach;
metoda read users przypisuje do pola users wartość poprzez wczytanie z pliku i następnie ją returnuje, a ta funkcja jest wykorzystywana w funkcji, która ją używa w środku addAll, na tym samym polu users, które również jest returnowane

Reply:
Ja pisałam tę metodę dla apki konsolowej; w tej chwili ciężko mi sobie przypomnieć, jaka mnie filozofia tu kierowała. Popatrzę na to, jak będę miała chwilę.